frontend/src/pages/modelServing/screens/projects/NIMServiceModal/NIMModelListSection.tsx

```diff
@@ -22,103 +21,107 @@ const NIMModelListSection: React.FC<NIMModelListSectionProps> = ({
+            const modelInfo = modelInfos.find((model) => model.name === modelName);
+            if (modelInfo && modelInfo.tags.length > 0) {
+              setSelectedModel(`${modelInfo.name}-${modelInfo.tags[0]}`);
+            } else {
+              setSelectedModel(modelName);
+            }
           }
-        } catch (err) {
-          setError('There was a problem fetching the NIM models. Please try again later.');
+        } else {
+          setError('No NVIDIA NIM models found. Please check the installation.');
           setOptions([]);
         }
-      };
-      getModelNames();
-    } else {
-      setOptions([
-        { key: inferenceServiceData.format.name, label: inferenceServiceData.format.name },
-      ]);
-      setSelectedModel(inferenceServiceData.format.name);
-    }
+      } catch (err) {
+        setError('There was a problem fetching the NIM models. Please try again later.');
+        setOptions([]);
+      }
+    };
+
+    getModelNames();
   }, [isEditing, inferenceServiceData.format.name]);
 
   const getSupportedModelFormatsInfo = (key: string) => {
     const lastHyphenIndex = key.lastIndexOf('-');
+    if (lastHyphenIndex === -1) {
+      return null;
+    }
     const name = key.slice(0, lastHyphenIndex);
     const version = key.slice(lastHyphenIndex + 1);
-
     const modelInfo = modelList.find((model) => model.name === name);
-    if (modelInfo) {
-      return {
-        name: modelInfo.name,
-        version,
-      };
-    }
-    return null;
+    return modelInfo ? { name: modelInfo.name, version } : null;
   };
 
   const getNIMImageName = (key: string) => {
     const lastHyphenIndex = key.lastIndexOf('-');
+    if (lastHyphenIndex === -1) {
+      return '';
+    }
     const name = key.slice(0, lastHyphenIndex);
     const version = key.slice(lastHyphenIndex + 1);
-
     const imageInfo = modelList.find((model) => model.name === name);
-    if (imageInfo) {
-      return `nvcr.io/${imageInfo.namespace}/${name}:${version}`;
+    return imageInfo ? `nvcr.io/${imageInfo.namespace}/${name}:${version}` : '';
+  };
+
+  const onSelect = (
+    _event: React.MouseEvent | React.KeyboardEvent | undefined,
+    key: string | number,
+  ) => {
+    if (typeof key !== 'string' || isEditing) {
+      return;
+    }
+    setSelectedModel(key);
+    const modelInfo = getSupportedModelFormatsInfo(key);
+    if (modelInfo) {
+      setServingRuntimeData('supportedModelFormatsInfo', modelInfo);
+      setServingRuntimeData('imageName', getNIMImageName(key));
+      setInferenceServiceData('format', { name: modelInfo.name });
+      setError('');
+    } else {
+      setError('Error: Model not found.');
     }
-    return '';
   };
 
   return (
     <FormGroup label="NVIDIA NIM" fieldId="nim-model-list-selection" isRequired>
-      <SimpleSelect
-        isScrollable
-        isFullWidth
+      <TypeaheadSelect
+        selectOptions={options}
+        selected={selectedModel}
```

Comment:
```suggestion
      <TypeaheadSelect
        selectOptions={options}
        selected={selectedModel}
        isScrollable
```

You should probably add the `isScrollable` prop on this. 

Currently the options will just keep going down the screen and get cut off
![image](https://github.com/user-attachments/assets/cfe2cf8a-6bc3-4902-8da2-9889a5f98e7e)

But with the prop it will let you scroll
![image](https://github.com/user-attachments/assets/16e05eae-2a62-4f39-9547-f9f73aeddffa)